source/common/common/macros.h

```diff
@@ -41,6 +43,15 @@ namespace Envoy {
     return *objectptr;                                                                             \
   } while (0)
 
+/**
+ * Assert memory bounds to avoid copy errors.
+ */
+#define SAFE_MEMCPY(dst, src)                                                                      \
+  do {                                                                                             \
+    ASSERT(sizeof(*(src)) == sizeof(*(dst)));                                                      \
```

Comment:
great; can you go back to static_assert then and see what fails?

Reply:
I will leave in place the static_assert and look for a work around for the reinterpret_cast.

Reply:
Can you paste the exact error message you are getting for the reinterpret_cast?

Reply:
Sorry for the late response 😓  I am unable to find the exact error in the past pipelines, but it was something like: 
`error: a reinterpret_cast is not a constant expression`
I willl try to avoid this error without changing the current definition of the macro with static_assert.